src/plugins/sessionPlayer/plugin.js

```diff
@@ -115,21 +116,58 @@ function processUpdatedSessions(instance, sessions, apiClient) {
     }
 }
 
-function getChangedEvents(state1) {
-    const names = [];
+function getChangedEvents(oldPlayerData, newPlayerData) {
+    let names = [];
 
-    if (!state1) {
+    if (!oldPlayerData?.PlayState || !newPlayerData?.PlayState
+        || (oldPlayerData.TranscodingInfo !== newPlayerData.TranscodingInfo && (!oldPlayerData.TranscodingInfo || !newPlayerData.TranscodingInfo))) {
         names.push('statechange');
-        names.push('timeupdate');
-        names.push('pause');
 
         return names;
     }
 
-    // TODO: Trim these down to prevent the UI from over-refreshing
-    names.push('statechange');
+    if (oldPlayerData.PlayState.PositionTicks !== newPlayerData.PlayState.PositionTicks) {
     names.push('timeupdate');
-    names.push('pause');
+    }
+    if (oldPlayerData.PlayState.IsPaused !== newPlayerData.PlayState.IsPaused) {
+        names.push(newPlayerData.PlayState.IsPaused ? 'pause' : 'unpause');
+    }
+    if (oldPlayerData.PlayState.IsMuted !== newPlayerData.PlayState.IsMuted
+        || oldPlayerData.PlayState.VolumeLevel !== newPlayerData.PlayState.VolumeLevel) {
+        names.push('volumechange');
+    }
+    if (oldPlayerData.PlayState.RepeatMode !== newPlayerData.PlayState.RepeatMode) {
+        names.push('repeatmodechange');
+    }
+
+    // oldPlayerData will be overwritten anyway
+    const prepareOldData = (oldObject, newObject, propertyName) => {
+        if (!Object.prototype.hasOwnProperty.call(newObject, propertyName)) {
+            delete oldObject[propertyName];
+        } else {
+            oldObject[propertyName] = newObject[propertyName];
+        }
+    };
+
+    prepareOldData(oldPlayerData.PlayState, newPlayerData.PlayState, 'PositionTicks');
+    if (oldPlayerData.TranscodingInfo) {
+        // TranscodingInfo.CompletionPercentage and TranscodingInfo.Framerate change with time
+        // so it's enough if we only trigger 'timeupdate' event
+        prepareOldData(oldPlayerData.TranscodingInfo, newPlayerData.TranscodingInfo, 'CompletionPercentage');
+        prepareOldData(oldPlayerData.TranscodingInfo, newPlayerData.TranscodingInfo, 'Framerate');
+    }
```

Comment:
Other fields from the "TranscodingInfo" object are not used in places where we watch player events, so maybe it is enough to overwrite the entire "TranscodingInfo" object here? To be safe, I only overwrote fields that I noticed that change over time. I'm not sure when "TranscodingInfo.BitRate" changes, but in the worst case, getChangedEvents will return statechange as it did before.